core/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala

```diff
@@ -506,6 +506,15 @@ case class SetTransaction(
   override def wrap: SingleAction = SingleAction(txn = this)
 }
 
+/**
+ * Stores the highest (inclusive) ID that has been assigned to a row in during history of the table.
+ * `preservedRowIds`, when set to true, indicates that all modified rows of this commit preserved
```

Comment:
What happens if we only modify rows actually, and the high water mark stays the same? We would not emit this action the way the iterator is implemented above, right? So also the commit would not have a RowIdHighWaterMark action indicating that preservation happened

Reply:
If you modify rows then you have a new `add` action without `baseRowId` for the newly written files containing the new rows (new fresh row ids always get assigned in that case)